lib/ui/painting/multi_frame_codec.cc

```diff
@@ -164,12 +177,13 @@ Dart_Handle MultiFrameCodec::getNextFrame(Dart_Handle callback_handle) {
 
   const auto& task_runners = dart_state->GetTaskRunners();
 
-  task_runners.GetIOTaskRunner()->PostTask(fml::MakeCopyable(
-      [callback = std::make_unique<DartPersistentValue>(
-           tonic::DartState::Current(), callback_handle),
-       this, trace_id, ui_task_runner = task_runners.GetUITaskRunner(),
-       io_manager = dart_state->GetIOManager()]() mutable {
-        GetNextFrameAndInvokeCallback(
```

Comment:
We can't use a WeakPtr here because we have to access the object across threads, and WeakPtr is not thread safe.

Reply:
And we can't create the weak ptr on the IO task runner because our test case would fail - e.g. if the IO runner is blocked, and the object gets collected before the task to create the weak ptr  runs